dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -679,15 +679,11 @@ public void run () {
         	contentlets = findContentletsByIdentifiers(identifiers, false, APILocator.getLanguageAPI().getDefaultLanguage().getId(), user, respectFrontendRoles);
         }
 
-        Map<String, Contentlet> map = new HashMap<String, Contentlet>(contentlets.size());
         for (Contentlet contentlet : contentlets) {
-            map.put(contentlet.getIdentifier(), contentlet);
-        }
-        for (String identifier : identifiers) {
-            if(map.get(identifier) != null && !contents.contains(map.get(identifier))){
-                contents.add(map.get(identifier));
-            }
+            if (identifierList.contains(contentlet.getIdentifier()))
```

Comment:
identifierList is a List and contains call method is O(N), you could change it to Log(N) just replacing the implementation to Set, it could be worthy to see if switch the collection implementation is a good idea.